src/main/java/org/mifos/connector/mpesa/camel/routes/SafaricomRoutesBuilder.java

```diff
@@ -201,6 +212,20 @@ public void configure() {
                     exchange.setProperty(TRANSACTION_ID, correlationId);
                 })
                 .process(collectionResponseProcessor)
+                .when(header("CamelHttpResponseCode").isEqualTo("500"))
+                .process(exchange -> {
+                    JSONObject jsonObject = new JSONObject(exchange.getIn().getBody(String.class));
+                    String errorCode = jsonObject.getString("errorCode");
+
+                    if (errorCode.equals("500.001.1001")) {
```

Comment:
As discussed, 500.001.1001 can mean different things so take the decision of pending txn based on contains logic and send a mail to API support to clarify whether this error code is specific to pending txn and request them for an error dictionary.

Reply:
```JSON
{
   "requestId":"ws_CO_03022022174912650571",
   "errorCode":"500.001.1001",
   "errorMessage":"The transaction is being processed"
}
```
The above JSON is the response for pending transactions and `contains` logic won't work because there is no word `pending` in the phrase.